### PR TITLE
✨ Add support for all `Field` parameters from Pydantic `1.9.0` and above, make Pydantic `1.9.0` the minimum required version

### PR DESCRIPTION
**Pull request by daniil-berg**
## Summary 

The proposed changes add support for the following `Field` parameters:
- `max_digits`
- `decimal_places`
- `unique_items`
- `discriminator`
- `repr`

This makes the SQLModel `Field` constructor support all [Pydantic field parameters](https://pydantic-docs.helpmanual.io/usage/schema/#field-customization) as of version `1.9.0` up until the current one (`1.10.2`).

Tests sub-package for Pydantic capabilities was added and tests for the new field parameters were included.

The Pydantic dependency was bumped to `1.9.0`.

## Fixes
- #354 - more elegant solution
- #356 - fulfills feature request

### CODE REVIEW

**Comment by rotemvilsa** [comment]
@tiangolo Is there any planned timeline to merge this PR? Discriminator is a need for many developers (that's why pydantic released a support for it). 
It could really make our lives way better as devs using SQLModel.

Thanks

**Comment by pajowu** [comment]
@tiangolo any update on this?

**Comment by N-Demir** [comment]
++ on updates

**Comment by tiangolo** [comment]
Awesome, thank you @daniil-berg! 🚀 

This will be available in the next version: `0.0.10` 🎉

File: tests/test_pydantic/test_field.py [hunk 1]
```diff
@@ -22,16 +22,6 @@ class Model(SQLModel):
...
-def test_unique_items():
-    class Model(SQLModel):
-        unique_strings: List[str] = Field(unique_items=True)
-
-    Model(unique_strings=["x", "y"])
-
-    with pytest.raises(ValidationError):
-        Model(unique_strings=["x", "y", "x"])
-
-
```
Comment (MRigal) [inline tests/test_pydantic/test_field.py]: why remove and not skipif, as pydantic v2 support is not present now? 

Please note that if that means Pydantic v2 support in few days, I'm fine with it ;-)

Comment (tiangolo) [reply tests/test_pydantic/test_field.py]: Yep, the plan is to support SQLAlchemy 2.0 and Pydantic 2.0 as soon as possible. 🤓

**Pull request merged**